### DISCUSSION TITLE
What does the E suffix stand for?

### DISCUSSION THREAD

**Discussion by mattdiamond** [gcanti/fp-ts#1671] [Q&A] [answered]
I noticed that `OptionT`, `EitherT`, and `TheseT` (along with their Task counterparts) have both `match` and `matchE` functions, but the Code Conventions FAQ doesn't cover the E suffix.

I can see that the `E` version uses chaining instead of mapping, so I understand the technical difference between these functions, but I'm just curious what word beginning with E is referenced here.

**Comment by gcanti** [comment, marked as answer]
IIRC "Effect"

**Comment by grossbart** [comment]
I added a brief explanation of the `E` suffix to https://gcanti.github.io/fp-ts/guides/code-conventions.html#what-an-e-suffix-means-eg-matche
